include/seqan3/alphabet/aminoacid/aminoacid_base.hpp

```diff
@@ -69,10 +69,23 @@ class aminoacid_base : public alphabet_base<derived_type, size, char>
                  !std::same_as<derived_type, other_aa_type> &&
                  aminoacid_alphabet<other_aa_type>
     //!\endcond
-    explicit constexpr aminoacid_base(other_aa_type const & other) noexcept
+    explicit constexpr aminoacid_base(other_aa_type const other) noexcept
     {
-        static_cast<derived_type &>(*this) =
-            detail::convert_through_char_representation<derived_type, other_aa_type>[seqan3::to_rank(other)];
+        if constexpr (is_constexpr_default_constructible_v<other_aa_type> &&
+    #if SEQAN3_WORKAROUND_GCC78_CONCEPT_ISSUES
+                      writable_alphabet<other_aa_type>
+    #else
+                      detail::writable_constexpr_alphabet<other_aa_type>
+    #endif // SEQAN3_WORKAROUND_GCC78_CONCEPT_ISSUES
+                     )
+        {
+            static_cast<derived_type &>(*this) =
+                detail::convert_through_char_representation<derived_type, other_aa_type>[seqan3::to_rank(other)];
+        }
+        else
+        {
+            assign_char_to(seqan3::to_char(other), static_cast<derived_type &>(*this));
```

Comment:
assign_char_to is a CPO cal. Because the name is different from the member (assign_char_to VS assign_char) it does not need to be prefixed with `seqan3::`.

Reply:
But I find it more expressive to always call CPOs qualified even in our own namespace. That reduces exactly these kind of possible confusions. Why is assign_char_to not prefixed but to_char is etc.